pkg/backup/compaction_job.go

```diff
@@ -610,24 +608,28 @@ func createCompactionManifest(
 	if len(tenantSpans) != 0 || len(tenantInfos) != 0 {
 		return nil, errors.New("backup compactions does not yet support range keys")
 	}
-	m, err := createBackupManifest(
-		ctx,
-		execCtx.ExecCfg(),
-		tenantSpans,
-		tenantInfos,
-		details,
-		compactChain.backupChain,
-		compactChain.allIters,
-	)
-	if err != nil {
-		return nil, err
-	}
-	m.IsCompacted = true
-	m.IntroducedSpans, err = compactIntroducedSpans(ctx, m, compactChain)
+	lastBackup := compactChain.lastBackup()
+	introducedSpans, err := compactChain.compactIntroducedSpans(ctx)
 	if err != nil {
 		return nil, err
 	}
-	return &m, nil
+	return &backuppb.BackupManifest{
+		StartTime:           details.StartTime,
+		EndTime:             details.EndTime,
+		MVCCFilter:          lastBackup.MVCCFilter,
+		Descriptors:         details.ResolvedTargets,
+		Tenants:             tenantInfos,
+		CompleteDbs:         lastBackup.CompleteDbs,
+		Spans:               lastBackup.Spans,
+		IntroducedSpans:     introducedSpans,
+		FormatVersion:       backupinfo.BackupFormatDescriptorTrackingVersion,
+		BuildInfo:           build.GetInfo(),
+		ClusterVersion:      execCtx.ExecCfg().Settings.Version.ActiveVersion(ctx).Version,
+		ClusterID:           execCtx.ExecCfg().NodeInfo.LogicalClusterID(),
```

Comment:
well, we're not actually backing up any state from the cluster running the compaction, so shouldn't we inherit the ID from lastBackup?